cpp/src/Ice/ConnectionI.cpp

```diff
@@ -2310,6 +2382,19 @@ Ice::ConnectionI::idleCheck(
     // else, nothing to do
 }
 
+void
+Ice::ConnectionI::inactivityCheck() noexcept
+{
+    // Called by the InactivityTimerTask.
+    std::lock_guard lock(_mutex);
+    if (isAtRest())
+    {
+        setState(
+            StateClosing,
+            make_exception_ptr(CloseConnectionException{__FILE__, __LINE__, "connection closed by inactivity timer"}));
```

Comment:
The user won't know about an inactivity timer. They just want to know why it's closed. 
```suggestion
            make_exception_ptr(CloseConnectionException{__FILE__, __LINE__, "connection closed due to inactivity"}));
```

We could maybe even add the time. `connection closed due to inactivity after X seconds`

Reply:
I updated the message.